search/includes/classes/class-search.php

```diff
@@ -814,6 +816,22 @@ public static function should_load_es_wp_query() {
 		return self::is_query_integration_enabled();
 	}
 
+	/**
+	 * Skip query integration if we don't have any active post indexes.
+	 *
+	 * @param Indexable $indexable Defaults to post.
+	 * @return bool Whether to skip query integration or not.
+	 */
+	public function maybe_skip_query_integration_if_no_active_indexes() {
+		$indexable = $this->indexables->get( 'post' );
+		if ( $indexable ) {
+			$version = $this->versioning->get_active_version_number( $indexable );
+			if ( is_wp_error( $version ) && 'no-active-version' === $version->get_error_code() ) {
```

Comment:
It would be cool if this included the mapping validity check, but I guess we can live without it.